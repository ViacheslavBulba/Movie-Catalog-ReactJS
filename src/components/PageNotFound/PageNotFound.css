.page-not-found-container {
    width: 100vw;
    min-height: calc(100vh - 70px);
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #232323;
}

.logo-page-not-found-container {
    width: 100vw;
    display: flex;
    flex-direction: row;
    justify-content: left;
    margin-top: 20px;
    margin-bottom: 20px;
    padding-left: 60px;
    margin-right: 60px;
}

.page-not-found-text {
    font-family: 'Source Sans Pro';
    font-size: 32px;
    color: #ffffff;
    margin-top: 5vh;
    margin-bottom: 5vh;
}

.not-found-image {
    height: 300px;
}

.go-back-button {
    width: 200px;
    height: 50px;
    border-radius: 5px;
    border: solid 1px #f65261;
    font-family: 'Source Sans Pro';
    font-size: 18px;
    color: #f65261;
    background-color: #232323;
    margin-top: 5vh;
    margin-bottom: 5vh;
}
