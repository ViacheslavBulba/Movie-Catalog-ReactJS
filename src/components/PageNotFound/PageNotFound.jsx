import React from 'react';
import './PageNotFound.css';
import logo from '../../../public/netflix-logo.svg';
import notFoundSvg from '../../../public/404-error.svg';
import Footer from '../Footer/Footer';
import { Link } from 'react-router-dom';
import { setMovieNotFoundById } from '../../store/actions';
import store from '../../store/store';

export default function PageNotFound() {
    const handleClick = () => {
        store.dispatch(setMovieNotFoundById(false));
    };

    return (
        <>
            <div className='page-not-found-container'>
                <div className='logo-page-not-found-container'>
                    <img src={logo} alt='logo' className='logo' />
                </div>
                <span className='page-not-found-text'>Page Not Found</span>
                <img
                    src={notFoundSvg}
                    alt='not found image'
                    className='not-found-image'
                />
                <Link to='/'>
                    <button className='go-back-button' onClick={handleClick}>
                        GO BACK TO HOME
                    </button>
                </Link>
            </div>
            <Footer />
        </>
    );
}
