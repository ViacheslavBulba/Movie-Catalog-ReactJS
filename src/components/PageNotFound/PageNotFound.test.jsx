import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import PageNotFound from './PageNotFound';

test('snapshot test for page not found', () => {
  const component = renderer.create(
    <MemoryRouter>
      <PageNotFound />
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
