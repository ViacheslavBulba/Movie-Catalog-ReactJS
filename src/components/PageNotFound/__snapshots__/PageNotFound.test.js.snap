// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for page not found 1`] = `
<div
  className="page-not-found-container"
>
  <div
    className="logo-page-not-found-container"
  >
    <img
      alt="logo"
      className="logo"
      src="netflix-logo.svg"
    />
  </div>
  <span
    className="page-not-found-text"
  >
    Page Not Found
  </span>
  <img
    alt="not found image"
    className="not-found-image"
    src="404-error.svg"
  />
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="go-back-button"
    >
      GO BACK TO HOME
    </button>
  </a>
</div>
`;
